### ISSUE TITLE
SpreadOperator rule flagging compliant code

### ISSUE THREAD

**Issue by shalbert94** [arturbosch/detekt#2012]
## Expected Behavior
Using Kotlin version `1.3.50`, the following should be compliant with the SpreadOperator rule 
```
protected fun getStringFromResource(@StringRes stringRes: Int, vararg arguments: Any) =
        itemView.context.getString(stringRes, *arguments)
```

## Observed Behavior
Detekt has flagged for violating the rule.

## Steps to Reproduce
To reproduce:
- Use Kotlin version `1.3.50`
- Turn on the `SpreadOperator` rule in your `config.yml` file
- Create a class
- Add the function above to that class
- Run Detekt
- The report will contain a violation of the `SpreadOperator` rule

## Your Environment

* Version of detekt used: `1.0.1`
* Version of Gradle used (if applicable): `3.3.0`
* Operating System and version: Android

**Comment by 3flex** [comment]
Couple of things to keep in mind:
1. This rule can return false positives when type resolution is not being used (e.g. when running detekt via `detektMain` task or similar instead of `detekt` task)
2. In case you're already doing that, the logic of the rule is taken directly from the Kotlin compiler and hasn't changed in two years, so the performance impact of the array copy still exists.

**Comment by shalbert94** [comment]
The `detekt` task was used in this case

**Comment by schalkms** [comment]
We comply with the Kotlin code analysis, since we directly call the referenced code.
Thus, I'll close this.
